sources/packages/api/src/services/atbc/atbc.controller.spec.ts

```diff
@@ -0,0 +1,97 @@
+require("../../../env_setup");
+import { closeDB, setupDB } from "../../testHelpers";
+import { Connection } from "typeorm";
+import * as faker from "faker";
+import { Student, User } from "../../database/entities";
+import { StudentService, ArchiveDbService, ATBCService } from "../../services";
+import { KeycloakConfig } from "../../auth/keycloakConfig";
+import { KeycloakService } from "../auth/keycloak/keycloak.service";
+import { ConfigService } from "../config/config.service";
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import { Test, TestingModule } from "@nestjs/testing";
+import * as request from "supertest";
+import { ATBCCreateClientResponse } from "../../types";
+
+describe("Test student model", () => {
+  const clientId = "student";
+  let connection: Connection;
+  let accesstoken: string;
+  let configService: ConfigService;
+  let app: INestApplication;
+
+  beforeAll(async () => {
+    await KeycloakConfig.load();
+    const token = await KeycloakService.shared.getToken(
+      process.env.E2E_TEST_STUDENT_USERNAME,
+      process.env.E2E_TEST_STUDENT_PASSWORD,
+      clientId,
+    );
+    accesstoken = token.access_token;
+
+    connection = await setupDB();
+    const moduleFixture: TestingModule = await Test.createTestingModule(
+      {},
+    ).compile();
+    app = moduleFixture.createNestApplication();
+    await app.init();
+  });
+  afterAll(async () => {
+    await closeDB();
+  });
+
+  it("should save student model object with user relationship and address jsonb", async () => {
+    // Create fake student in SIMS DB
+    const archiveDB = new ArchiveDbService();
+    const studentService = new StudentService(connection, archiveDB);
+    const atbcService = new ATBCService(configService, studentService);
+    const fakestudent = new Student();
+    fakestudent.sin = "123456789";
+    fakestudent.birthdate = faker.date.past(20);
+    fakestudent.gender = "F";
+    fakestudent.contactInfo = {
+      addresses: [
+        {
+          addressLine1: faker.address.streetAddress(),
+          city: faker.address.city(),
+          country: "Canada",
+          province: "ON",
+          postalCode: faker.address.zipCode(),
+        },
+      ],
+      phone: faker.phone.phoneNumber(),
+    };
+    const simsUser = new User();
+    simsUser.userName = faker.random.uuid();
+    simsUser.email = faker.internet.email();
+    simsUser.firstName = faker.name.firstName();
+    simsUser.lastName = faker.name.lastName();
+    fakestudent.user = simsUser;
+
+    // Save the student in SIMS
+    await studentService.save(fakestudent);
```

Comment:
Here the student is being saved but the "getStudentByUserName" is also being mocked. Are both being used?
I believe that the mocked method will make the database not be used at all.

Reply:
`getStudentByUserName` is mocked because the username is from BCSC, the fake user created in the DB doesn't have a username from BCSC, that is why it is mocked. The fake user in DB is created to update the PD sent date, `studentPDSentAt` column

Reply:
The user name on token is supposed to be "student_e2e_test". If the fake user name is created using this username it should be aligned with what the UserToken will be receiving. Does it make sense?
The token, in the last instance has the "key cloack user name and info" not BCSC. Whet happens is, when the user is authenticated using BCSC, a Key Clock user is created using BCSC info.